com.unity.shadergraph/Editor/Data/Graphs/GraphData.cs

```diff
@@ -1367,16 +1398,23 @@ public string SanitizeGraphInputReferenceName(ShaderInput input, string desiredN
             {
                 case AbstractShaderProperty property:
                 {
-                    // must deduplicate ref names against both keyword and properties, as they occupy the same name space
-                    var existingNames = properties.Where(p => p != property).Select(p => p.referenceName).Union(keywords.Select(p => p.referenceName));
+                    // must deduplicate ref names against keywords, dropdowns, and properties, as they occupy the same name space
+                    var existingNames = properties.Where(p => p != property).Select(p => p.referenceName).Union(keywords.Select(p => p.referenceName)).Union(dropdowns.Select(p => p.referenceName));
```

Comment:
yikes.  we need a better way to do unique names in the future.  :)

Reply:
Yes, this is a bit rough! A helper function would simplify this for sure.